server/CGameHandler.cpp

```diff
@@ -993,7 +993,7 @@ void CGameHandler::applyBattleEffects(BattleAttack &bat, const CStack *att, cons
 		StacksHealedOrResurrected::HealInfo hi;
 		hi.stackID = att->ID;
 		hi.healedHP = bsa.killedAmount * att->valOfBonuses(Bonus::SOUL_STEAL) * att->MaxHealth(); //TODO: Should unit be additionally healed after life drain?
-		hi.lowLevelResurrection = false;
+		hi.lowLevelResurrection = att->hasBonusOfType(Bonus::SOUL_STEAL, 1);
```

Comment:
F.e. In this case creature may have 25% soul steal permanently + 25% till the end of battle.